ros-rviz-point-cloud-2.html

```diff
@@ -73,16 +87,40 @@
           return;
         }
         this.destroy();
+        this._rgb_lut = new Float32Array(256);
+        for (var i = 0; i < 256; i++) {
+          this._rgb_lut[i] = i / 255.0
+        }
+        this._floatColor = new Float32Array(1);
         var that = this;
+
+        let colormap = undefined;
+        if (this.applyRGB8PCT) {
+          colormap = function(x) {
+            that._floatColor[0] = x;
+            const intColorArray = new Int32Array(that._floatColor.buffer);
+            const intColor = intColorArray[0];
+
+            return {
+              r: that._rgb_lut[(intColor >> 16) & 0xff],
+              g: that._rgb_lut[(intColor >> 8) & 0xff],
+              b: that._rgb_lut[(intColor) & 0xff],
```

Comment:
Instead of converting the float to an int using arrays and applying masks, I also tried creating a `Uint8Array` from the float array, and picking the bytes I need for each channel.

While that would be a bit more efficient, it wouldn't work with different word-endian. This transformation in principle doesn't know if the data is big or little endian. While we could find a way to route the endian type here, IMO this is a better compromise solution.